### ISSUE TITLE
About the implementation

### ISSUE THREAD

**Issue by tholmber** [rpautrat/SuperPoint#162]
Hi, First of all, thanks for sharing the great project!
My questions are more about the implementation of SuperPoint in general than some bug or error.

1) I have trained MagicPoint model several times with pseudo-labeled MS-COCO data and currently everything is working as expected. After one self-training iteration, the training part takes only 1.5 hours but pseudo-labeling for MS-COCO data takes 28 hours. Does it sound correct? I also tested by increasing the batch size but there were no big impact on prediction time of pseudo-labels.

2) I assume that since MS-COCO (training set) contains 80 000 samples, there are no way to read files from the data folder and write pseudo-labels faster to destination folder. If I understood correctly, np.savez_compressed() saves predicted keypoints in numpy array in .npz format. How things would change if data and pseudo-labels are saved as tfrecords? Probably it demands in addition Tensorflow 2 implementation of SuperPoint but I'm curious to know your opinion about this imaginary scenario.

Thanks

**Comment by rpautrat** [comment]
Hi,

1. Yes, that's correct. Writing the npz files takes a lot of time unfortunately...

2. I haven't tried to use tfrecords instead of the npz files, but this could be tried indeed. Not sure if it would really be faster, you might want to do a small comparison first to check if it would really bring some improvement.

**Comment by tholmber** [comment]
Thanks for the fast reply! I will do small tests and open this one again if I get good results.

**Comment by tholmber** [comment]
I tried to save keypoints as tfrecords and it was even slower than writing npz files. I just need to accept that MS-COCO is big dataset and it takes time to generate pseudo-labels :)

**Comment by aniket-gupta1** [comment]
Hey, So I am also facing the same issue and I tried measuring the time taken for writing .npz files. It appears to me that writing the files is not the issue. 
Instead the code takes a lot of time in predicting the keypoint maps. 

[https://github.com/rpautrat/SuperPoint/blob/1742343e7a9731929be00f7a940023b694098d6c/superpoint/export_detections.py#L62]

This line takes about 2 seconds for a batch size of 5. Do you have any ideas why that might be happening?

**Comment by rpautrat** [comment]
Hi, the process of homography adaptation is quite slow, because it runs 100 forward passes for each image. Thus, there is no easy way to make it faster.
One could reduce the number of homographies to less than 100, but this would also reduce a bit the quality of the pseudo ground truth.
